src/v/storage/parser_utils.h

```diff
@@ -13,9 +13,20 @@
 
 #include "bytes/iobuf_parser.h"
 #include "model/record.h"
+#include "model/record_batch_reader.h"
 
 namespace storage::internal {
 
+/// \brief Decompress over a model::record_batch_reader
+class decompress_batch_consumer {
```

Comment:
doesn't the record_batch already have something like this `batch.for_each_record` ?

Reply:
nice anyway.

Reply:
No that actually throws an assertion if the batch is not already decompressed